java/templates/s2i/run

```diff
@@ -13,6 +13,45 @@ JAVA_OPTIONS="${JAVA_OPTIONS:+${JAVA_OPTIONS} }$(/opt/jolokia/jolokia-opts)"
 JVM_ARGS="${JVM_ARGS:+${JVM_ARGS} }${JAVA_OPTIONS}"
 export JAVA_OPTIONS JVM_ARGS
 
+# support JBoss Tools OpenShift debug environment variables
+export JAVA_DEBUG="${JAVA_DEBUG:-${DEBUG}}"
+if [ "x${JAVA_DEBUG}" != x ]; then
+  export JAVA_DEBUG_PORT="${JAVA_DEBUG_PORT:-${DEBUG_PORT}}"
+fi
+# SpringBoot also uses DEBUG variable, so shade it with SPRINGBOOT_DEBUG variable
+if [ "x${SPRINGBOOT_DEBUG}" != x ]; then
+	export DEBUG="${SPRINGBOOT_DEBUG}"
+else
+	unset DEBUG
+fi
+
+# check for SpringBoot exploded archive
+if [ -f "${DEPLOYMENTS_DIR}/META-INF/MANIFEST.MF" ]; then
+	echo "SpringBoot exploded archive detected..."
+	if [ "x${JAVA_DEBUG}" != x ]; then
+		# does archive container spring devtools in lib
+		devtools_present=`ls "${DEPLOYMENTS_DIR}"/BOOT-INF/lib/spring-boot-devtools-*.jar 2>/dev/null | wc -l | tr -d '[[:space:]]'`
+		if [ ${devtools_present} = 1 ]; then
+			echo "SpringBoot developer tools library detected..."
+			# setup classpath and start class as main class
+			if [ "x${JAVA_CLASSPATH}" = x ]; then
+				JAVA_CLASSPATH="${DEPLOYMENTS_DIR}/BOOT-INF/lib/*"
+			else
+				JAVA_CLASSPATH="${DEPLOYMENTS_DIR}/BOOT-INF/lib/*:${JAVA_CLASSPATH}"
+			fi
+			export JAVA_CLASSPATH="${DEPLOYMENTS_DIR}/BOOT-INF/classes/:${JAVA_CLASSPATH}"
+			# setup start class as main class
+			main_class=`sed -n 's/Start-Class: \(.*\)$/\1/p' "${DEPLOYMENTS_DIR}/META-INF/MANIFEST.MF" | tr -d '[[:space:]]'`
+		fi
+	fi
+	# default to SpringBoot's Main-Class
+	if [ "x${main_class}" = x ]; then
```

Comment:
ok, convinced :) It's probably better to keep run-java.sh tech agnostic.

I'm not that deep in how spring devtools work actually, but I know that they need special jars prepared for working properly. Therefore we do also a new repackaging in f-m-p when we do hot deploy support for spring boot in `fabric8:watch`. I'm not super happy with this because it creates different artifacts (with the same 'version' number) and it might happen accidentally that devtools slip into production (e.g if there's not `mvn clean` somewhere inbetween).

would have to revisit the fmp code more closely to make a better statement, but I'm perfetlu fine with this PR for now.

Just lets update the README and then merge.